### ISSUE TITLE
Keras saved model returns different result than original model using Batch Normalization on multiple GPUs (Distributed training)

### ISSUE THREAD

**Issue by Lucasbravo1** [keras-team/keras#14372]
I got a different result when using evaluate function on a saved model when compare with the original model. This only happens when Batch Normalization is included in the model and when training on multiple GPUs with MirroredStrategy.

Here is my model
```
with strategy.scope():
  model = tf.keras.Sequential([
      tf.keras.layers.Conv2D(32, 3, activation='relu', input_shape=(28, 28, 1)),
      tf.keras.layers.MaxPooling2D(),
      tf.keras.layers.BatchNormalization(),
      tf.keras.layers.Flatten(),
      tf.keras.layers.Dense(64, activation='relu'),
      tf.keras.layers.Dense(10)
  ])

  model.compile(loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
                optimizer=tf.keras.optimizers.Adam(),
                metrics=['accuracy'])
```

Multiple GPUs with MirroredStrategy
```
strategy = tf.distribute.MirroredStrategy()
print('Number of devices: {}'.format(strategy.num_replicas_in_sync))
```
Output
```
Number of devices: 2
```

Evaluate after training
```
eval_loss, eval_acc = model.evaluate(eval_dataset)

print('Eval loss: {}, Eval Accuracy: {}'.format(eval_loss, eval_acc))
```

Output
```
79/79 [==============================] - 0s 5ms/step - loss: 0.0424 - accuracy: 0.9884
Eval loss: 0.04239395260810852, Eval Accuracy: **0.9883999824523926**
```

Save model and evaluate again
```
path = 'saved_model/'
model.save(path, save_format='tf')
with strategy.scope():
  replicated_model = tf.keras.models.load_model(path)
  replicated_model.compile(loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
                           optimizer=tf.keras.optimizers.Adam(),
                           metrics=['accuracy'])

  eval_loss, eval_acc = replicated_model.evaluate(eval_dataset)
  print ('Eval loss: {}, Eval Accuracy: {}'.format(eval_loss, eval_acc))
```
Output
```
79/79 [==============================] - 0s 6ms/step - loss: 0.0424 - accuracy: 0.9883
Eval loss: 0.04239019751548767, Eval Accuracy: **0.9883000254631042**
```

 - Without BN

Output from evaluate
```
79/79 [==============================] - 0s 6ms/step - loss: 0.0450 - accuracy: 0.9837
Eval loss: 0.04498908668756485, Eval Accuracy: **0.9836999773979187**
```

Save model and repeat evaluate
```
79/79 [==============================] - 0s 4ms/step - loss: 0.0450 - accuracy: 0.9837
Eval loss: 0.04498908668756485, Eval Accuracy: **0.9836999773979187**
```

I have searched for similar issues and thought that this is because [BN computes differently during training and interference ](https://keras.io/api/layers/normalization_layers/batch_normalization/) but when I tried with one GPU, this issue didn't occur.

These are tested on Tensorflow 2.3.1, Keras 2.4.3 and CUDA 10.1

[Code for reproduce results](https://colab.research.google.com/drive/14iv88UJwFSv1SaVzmEl9RMOa1EH8IiRG?usp=sharing)

**Comment by victor-sm** [comment]
I have the same problem, the only "solution" so far was to turn off all but one GPU:

```
import tensorflow as tf
tf.config.get_visible_devices()
gpus = tf.config.experimental.list_physical_devices('GPU')
tf.config.experimental.set_visible_devices(gpus[0], 'GPU')
```

There seems to be something fundamentally broken with the BatchNormalization-Layer or the SavedModel. I encountered this error using a Azure GPU Instance with:

2 and 4 GPUs
Python 3.6 and 3.8
Tensorflow 2.3, 2.4 and 2.6

**Comment by sushreebarsa** [comment]
@Lucasbravo1 In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
